src/knx/group_object.cpp

```diff
@@ -279,3 +279,30 @@ void GroupObject::valueNoSend(const KNXValue& value, const Dpt& type)
 
     KNX_Encode_Value(value, _data, _dataLength, type);
 }
+
+bool GroupObject::valueModifiedSend(const KNXValue& value, const Dpt& type, const bool forceSend /*= false*/)
+{
+    if (_commFlag == Uninitialized)
+    {
+        // always send first value
+        this->value(value, type);
+        return true;
+    }
+    else
+    {
+        // convert new value to given dtp
+        uint8_t newData[_dataLength];
+        KNX_Encode_Value(value, newData, _dataLength, type);
+
+        // check for change in converted value / update value on change only
+        const bool dataChanged = memcmp(_data, newData, _dataLength);
+        if (dataChanged)
+            memcpy(_data, newData, _dataLength);
+
+        // trigger sending
+        if (dataChanged || forceSend)
```

Comment:
Remove this if and execute objectWritten() directly after memcpy.

Reply:
Would change (intended) behaviour: The return value should always indicate a changed value/state, independent of writing to bus.